modules/data-streams/src/main/java/org/elasticsearch/datastreams/DataStreamFeatures.java

```diff
@@ -25,6 +25,7 @@
 public class DataStreamFeatures implements FeatureSpecification {
 
     public static final NodeFeature DATA_STREAM_LIFECYCLE = new NodeFeature("data_stream.lifecycle");
+    public static final NodeFeature DATA_STREAM_INDICES_EXTRACTION = new NodeFeature("data_stream.indices_extraction");
```

Comment:
Oof, this is a tricky one. I can see the following two scenarios:

**Node feature independent on feature flag**
We add the node feature just like it is added here.

Analysis:
+ There will be no problems on forming clusters on tests.
+ For tests to work we will need the feature flag to be on and the node feature to be present (probably already the case, that's why is not a minus).

- The node feature exists on both snapshot & release versions (aka exposed to our users)
- When going GA, if we need a node feature we will need to add another one because this one will pre-exist the release, it will be present before the feature exists.

If we keep this, I would suggest we settle on adding something in the name to denote that it does not reflect a feature, for example, `internal.data_stream.indices_extraction` or `test_only.data_stream.indices_extraction`. 

**Node feature dependent on feature flag**
We create the node feature but we only return it if the feature flag is enabled. Preferably denote this in the name or with a comment.  

Analysis:
+ The node feature is not leaked to the release versions and our users.
+ When going GA, we can introduce a single node feature that will capture all the changes.

- We need to ensure that all tests are enabling the feature flag because we have mixed cluster tests, the release versions will not include the node feature and it will break our clusters. So, there will be some work that needs to be done here.

Let me know if I am missing anything. I am leaning towards the second, it's more work but it feels cleaner, what if we need to add new node features later while still behind a feature flag, I am afraid things will get really polluted. I do not feel very strong though, given that we have a naming convention to separate node features that are "user facing" and ones that are not.

Any thoughts?

Reply:
I have confirmation that the docs I found were inaccurate and that `cluster_features: "gte_v8.15.0"` was added as a stop gap so we can make use of it on here. I don't know if we'll be able to get a clean answer for feature flags and Node Feature usage for the time being, considering feature flag usage is already pretty nebulous. In the future, this kind of thing would be controlled by the new Capabilities API that's being developed. Those entries can be changed over time unlike the Node Feature API.

Reply:
All right, so we're going with `cluster_features: "gte_v8.15.0"` for now then?